browser/TemplateProvider.js

```diff
@@ -30,60 +30,44 @@
 
 'use strict';
 
-module.exports = TemplateProvider;
-
-/**
- * Creates new instance of Handlebars template provider.
- * @param {Handlebars} $handlebars Handlebars factory.
- * @constructor
- */
-function TemplateProvider($handlebars) {
-	this._handlebars = $handlebars;
-	this._templates = Object.create(null);
-}
+class TemplateProvider {
+	constructor(locator) {
+		this._handlebars = locator.resolve('handlebars');
+		this._templates = Object.create(null);
+	}
 
-/**
- * Current Handlebars factory.
- * @type {Handlebars}
- * @private
- */
-TemplateProvider.prototype._handlebars = null;
+	/**
+	 * Registers compiled (precompiled) Handlebars template.
+	 * http://handlebarsjs.com/reference.html
+	 * @param {string} name Template name.
+	 * @param {string} compiled Compiled template source.
+	 */
+	registerCompiled(name, compiled) {
 
-/**
- * Current set of registered templates.
- * @type {Object}
- * @private
- */
-TemplateProvider.prototype._templates = null;
+		/* eslint no-new-func: 0 */
+		var specs = new Function(`return ${compiled};`);
+		this._templates[name] = this._handlebars.template(specs());
+	}
 
-/**
- * Registers compiled (precompiled) Handlebars template.
- * http://handlebarsjs.com/reference.html
- * @param {string} name Template name.
- * @param {string} compiled Compiled template source.
- */
-TemplateProvider.prototype.registerCompiled = function (name, compiled) {
-	// jshint evil:true
-	var specs = new Function('return ' + compiled + ';');
-	this._templates[name] = this._handlebars.template(specs());
-};
+	/**
+	 * Renders template with specified data.
+	 * @param {string} name Name of template.
+	 * @param {Object} data Data context for template.
+	 * @returns {Promise<string>} Promise for rendered HTML.
+	 */
+	render(name, data) {
+		if (!(name in this._templates)) {
+			return Promise.reject(new Error('No such template'));
```

Comment:
It would be better to include the template's name to the error message. `"${name}" not found among registered templates`